pystiche_papers/johnson_alahi_li_2016/_utils.py

```diff
@@ -42,10 +42,30 @@ def postprocessor() -> transforms.CaffePostprocessing:
 
 
 def multi_layer_encoder(impl_params: bool = True,) -> enc.VGGMultiLayerEncoder:
+    r"""Multi-layer encoder from :cite:`JAL2016`.
+
+    Args:
+        impl_params: If ``True``, use the parameters used in the reference
+            implementation of the original authors rather than what is described in
+            the paper. For details see below.
+
+    If ``impl_params`` is ``True`` , an external instead of an internal preprocessing of
+    the images is used.
+    """
     return enc.vgg16_multi_layer_encoder(
         weights="caffe", internal_preprocessing=not impl_params, allow_inplace=True
     )
 
 
 def optimizer(transformer: nn.Module) -> optim.Adam:
+    r"""Optimizer from :cite:`JAL2016`.
+
+    Args:
+        transformer: Transformer to be optimized.
+
+    Returns:
+        :class:`torch.optim.Adam` optimizer with a learning rate of ``1e-3``. The
+        parameters of ``transformer`` are set as optimization parameters.
```

Comment:
The type of the output is added automatically.

```suggestion
        Optimizer for the ``transformer`` parameters.
```